astropy/modeling/mappings.py

```diff
@@ -182,3 +183,117 @@ def inverse(self):
         """
 
         return self
+
+
+class UnitsMapping(Model):
+    """
+    Coerce input units (or dimensionless input) into new units without
+    conversion.  Used by Model.coerce_units to support units on otherwise
```

Comment:
>  Coerce input units (or dimensionless input) into new units without conversion.

This is particularly difficult to imagine. I think it needs a little bit of re-wording, in particular the _"without conversion"_ part. Also, how does this relate to the examples: `model(units.Quantity(10, units.m))` and `model(units.Quantity(10, units.cm))`. I would like to be able to know what to expect from these examples after reading this paragraph.

Reply:
My description is unclear, I'll try reworking it

Reply:
(updated this)